### PR TITLE
Rename scala.runtime.{Tuple => Tuples}

### PR DESCRIPTION
**Pull request by smarter**
To avoid confusion with scala.Tuple. Part of #10602.

### CODE REVIEW

File: library/src-bootstrapped/scala/Tuple.scala [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  /** Converts a tuple `(T1, ..., Tn)` to a flattened `(..F[T1], ..., ..F[Tn])` */
+  type FlatMap[Tup <: Tuple, F[_] <: Tuple] <: Tuple = Tup match {
+    case EmptyTuple => EmptyTuple
+    case h *: t => Concat[F[h], FlatMap[t, F]]
+  }
+
+  /** Filters out those members of the tuple for which the predicate `P` returns `false`.
+   *  A predicate `P[X]` is a type that can be either `true` or `false`. For example:
+   *  ```
+   *  type IsString[x] = x match {
```
Comment (prolativ) [inline library/src-bootstrapped/scala/Tuple.scala:130]: As pointed out here https://github.com/lampepfl/dotty/issues/10896 this example is misleading and won't work without a type bound
```scala 
type IsString[x] <: Boolean = x match {
```

Comment (smarter) [reply library/src-bootstrapped/scala/Tuple.scala:130]: Fixing that is independent of this PR.

**Comment by nicolasstucki** [comment]
We should not lose the git history of those sources. We can achieve this the same way I did in #10998.

**Comment by som-snytt** [comment]
It would be more stylish and distinctive if `SCala.Runtime.Tuples` were called `Scruples`.

**Comment by smarter** [comment]
@nicolasstucki it looks like github rename detection in the changes view doesn't handle multiple copies of the same file, but git does: if I do `git log --follow library/src/scala/runtime/Tuples.scala`, the full history is displayed, so I don't think there's anything more to do. it's true that if I do `git show` on the commit it's not displayed as a copy, bu this is fixable by running `git config --global diff.renames copies`

**Comment by nicolasstucki** [comment]
We cannot use `git log --follow` within GitHub, hence I'm not convinced that is a useful workaround.

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**